teps/0027-https-connection-to-triggers-eventlistener.md

```diff
@@ -0,0 +1,135 @@
+---
+title: https-connection-to-triggers-eventlistener
+authors:
+  - "@savitaashture"
+creation-date: 2020-10-19
+last-updated: 2020-10-19
+status: proposed
+---
+
+# TEP-0024: HTTPS Connection To Triggers EventListener
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories](#user-stories)
+  - [Usage examples](#usage-examples)
+- [Design Details](#design-details)  
+- [Upgrade &amp; Migration Strategy](#upgrade--migration-strategy)
+- [References](#references)
+<!-- /toc -->
+
+## Summary
+
+This TEP proposes to provide flexibility to configure triggers eventlistener to support 
+`HTTPS` connection along with existing `HTTP` connection.
+
+## Motivation
+
+Providing inbuilt feature to support secure connection to eventlistener helps to avoid usage of any thirdparty solution
+and also helps trigger users to configure their events securely with few additional configurations. 
+
+### Goals
+
+* Make an additive change to the EventListener API to support `HTTPS` connection.
+
+### Non-Goals
+
+* Creating and maintaining certificates which are required for `HTTPS` connection.
+
+## Proposal
+
+Triggers support end to end secure `HTTPS` connection to eventlistener pod along with existing `HTTP` connections.
+
+### User Stories
+
+* User want to replace parts of their (Jenkins) build infrastructure with a cloud-native build infrastructure,
+and their security requirements specify that TLS needs to be used for both communication outside the cluster and 
+inside the cluster then having this solution would help. 
+
+* Users using custom/inbuilt interceptors like Github want their request to have secure connection all the way to the tekton trigger eventlistener pod then that can be solved with this feature.
+
+### Usage examples
+
+```yaml
+apiVersion: triggers.tekton.dev/v1alpha1
+kind: EventListener
+metadata:
+  name: github-listener-interceptor
+spec:
+  triggers:
+    - name: github-listener
+      interceptors:
+        - github:
+            secretRef:
+              secretName: github-secret
+              secretKey: secretToken
+            eventTypes:
+              - pull_request
+      bindings:
+        - ref: github-pr-binding
+      template:
+        name: github-template
+  resources:
+    kubernetesResource:
+      spec:
+        template:
+          spec:
+            serviceAccountName: tekton-triggers-github-sa
+            volumes:
+              - name: https-connection
+                secret:
+                  secretName: ssh-key-secret
+            containers:
+            - volumeMounts:
+                - name: https-connection
+                  mountPath: "/etc/triggers/ssl/"
+                  readOnly: true
+              env:
+              - name: SSL_CERT_DIR
+                value: "/etc/triggers/ssl/tls.crt"
+              - name: SSL_KEY_DIR
+                value: "/etc/triggers/ssl/tls.key"
```

Comment:
`SSL_CERT_DIR` should specify a folder (or a group of them), see [here](https://golang.org/src/crypto/x509/root_unix.go)